index.js

```diff
@@ -20,36 +23,74 @@ const resolveAliace = (str, alias) => {
     str = str.replace(re, (...f) => {
       const fk = key => new RegExp(`${key}/`);
       let findedKey = entries(alias).find(([key]) => fk(key).test(f[2]));
-      const base = findedKey[1];
-      const add = f[2].replace(fk(findedKey[0]), '');
-      const path = join(base, add);
-      const ret = stringify(path).replace(/"/gim, '');
-      return ret;
+      if(findedKey) {
+        const base = findedKey[1];
+        const add = f[2].replace(fk(findedKey[0]), '');
+        const path = join(base, add);
+        const ret = stringify(path).replace(/"/gim, '');
+        return ret;
+      } else {
+        return ``;
+      }
     });
   }
   return str;
 };
 
-const stdProcessArray = alias => {
+const replaceProcess = (str, replaceArray) => {
+  for (let [key, value] of entries(replaceArray)) str = str.replace(new RegExp(k`${key}`, 'igm'), value);
+  return str;
+};
+
+const stdProcessArray = ({alias, replace}) => {
   if(alias && objectLength(alias) > 0) {
     return [
       str => resolveAliace(str, alias),
+      str => replaceProcess(str, replace),
       str => clearAllKey(str)
     ];
   }
   return [];
 };
 
 const transformPlugin = ctx => {
-  const params = { alias: { }, callbackArray: [ ], ...ctx };
-  const { alias, callbackArray } = params;
+  const params = {
+    alias: { },
+    replace: { },
+    callbackArray: [ ],
+    exclude: [ ],
+    tStart,
+    tEnd,
+    ...ctx
+  };
+
+  tStart = params.tStart;
+  tEnd = params.tEnd;
+
+  const {
+    alias,
+    replace,
+    exclude,
+    callbackArray
+  } = params;
+
+  const checkExcludeStatus = id => {
+    for (let excludePattern of exclude) if(new RegExp(excludePattern, 'igm').test(id)) return true
+    return false
+  }
+
   return {
     name: 'transformPlugin',
-    transform: ctx => pipe(
-      str => str, // no idea!
-      ...stdProcessArray(alias),
-      ...callbackArray
-    )(ctx)
+    transform: (ctx, id) => {
+      const flag = checkExcludeStatus(id);
+      console.log(flag, id);
```

Comment:
Is this log necessary? Did you forget to delete the debug log?

Reply:
Indeed it is. Thank you. Fixed in the new version.

Reply:
Indeed so. Thanks. corrected in the new version.

Sorry, I didn't understand the capabilities of the github. it might be possible to add your code suggestion, but I did not understand how, and made the changes directly.